fabric-client-tags-api-v1/src/client/java/net/fabricmc/fabric/api/tag/client/v1/ClientTags.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tag.client.v1;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.tag.TagKey;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryEntry;
+import net.minecraft.util.registry.RegistryKey;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.tag.client.ClientTagsLoader;
+
+/**
+ * Allows the use of tags by directly loading them from the installed mods.
+ *
+ * <p>Tags are loaded by the server, either the internal server in singleplayer or the connected server and
+ * synced to the client. This can be a pain point for interoperability, as a tag that does not exist on the server
+ * because it is part of a mod only present on the client will no longer be available to the client that may wish to
+ * query it.
+ *
+ * <p>Client Tags resolve that issue by lazily reading the tag json files within the mods on the side of the caller,
+ * directly, allowing for mods to query tags such as {@link net.fabricmc.fabric.api.tag.convention.v1.ConventionalBlockTags}
+ * even when connected to a vanilla server.
+ */
+@Environment(EnvType.CLIENT)
```

Comment:
Just a note you dont actually need this when using split sourcesets. Loom automatically adds it to the re-merged jar. Nothing wrong with leaving it here tho.